BASE/Test/ServerTelemetryChannel.Test/Shared.Tests/Implementation/TelemetryBufferTest.cs

```diff
@@ -379,6 +380,137 @@ public void DoesNotContinueOnCapturedSynchronizationContextToImprovePerformance(
             }
         }
 
+        [TestClass]
+        public class ManualFlushAsync : TelemetryBufferTest
+        {
+            [TestMethod]
+            [Timeout(10000)]
+            public void DoesntSerializeTelemetryIfBufferIsEmpty()
+            {
+                bool telemetrySerialized = false;
+                var serializer = new StubTelemetrySerializer();
+                serializer.OnSerialize = telemetry =>
+                {
+                    telemetrySerialized = true;
+                };
+                var telemetryBuffer = new TelemetryBuffer(serializer, new StubApplicationLifecycle());
+
+                telemetryBuffer.ManualFlushAsync(new CancellationToken()).GetAwaiter().GetResult();
```

Comment:
Nit: use `CancellationToken.None` or, if you're allowed to use newer C# langauge features, just `default`